persistent-postgresql/Database/Persist/Postgresql.hs

```diff
@@ -1320,24 +1343,33 @@ showAlter table (Update' c s) = T.concat
     ]
 showAlter table (AddReference reftable fkeyname t2 id2 cascade) = T.concat
     [ "ALTER TABLE "
-    , escapeE table
+    , escapeS table
     , " ADD CONSTRAINT "
     , escapeC fkeyname
     , " FOREIGN KEY("
     , T.intercalate "," $ map escapeF t2
     , ") REFERENCES "
-    , escapeE reftable
+    , escapeS reftable
     , "("
     , T.intercalate "," id2
     , ")"
     ] <> renderFieldCascade cascade
 showAlter table (DropReference cname) = T.concat
     [ "ALTER TABLE "
-    , escapeE table
+    , escapeS table
     , " DROP CONSTRAINT "
     , escapeC cname
     ]
 
+schemaNamePair :: EntityDef -> SchemaEntityName
+schemaNamePair ent = schemaNamePair' (getEntityDBSchema ent) (getEntityDBName ent)
+
+schemaNamePair' :: Maybe Text -> EntityNameDB -> SchemaEntityName
+schemaNamePair' mbSchema entName = case mbSchema of
+    Nothing -> ("", entName)
+    Just "public" -> ("", entName)
+    Just schema -> (schema, entName)
```

Comment:
Uses of this function case on `""` and treat it differently - I think that's a good reason to keep this as a `Maybe Text` instead of a `Text`.

I'm also curious about treating `"public"` differently here - can you explain why we don't just keep the `public` prefix?  Presumably, the user must have specified that - ie `Foo schema=public` - and I'm hesitant to override a user choice.